spyder/plugins/switcher/plugin.py

```diff
@@ -41,6 +46,50 @@ class Switcher(SpyderPluginV2):
     CONF_FILE = False
 
     # --- Signals
+    sig_rejected = Signal()
+    """
+    This signal is emitted when the plugin is dismissed.
+    """
+
+    sig_text_changed = Signal(TEXT_TYPES[-1])
```

Comment:
```suggestion
    sig_text_changed = Signal(str)
```

We can safely use `str` here because we don't support Python 2 anymore.